external/KkthnxUI/Core.KkthnxUI.lua

```diff
@@ -1,68 +1,59 @@
 local ADDON_NAME, namespace = ...
 local L = namespace.L
+local Version = nil
 
-local cargBags = nil
 local addonName = "KkthnxUI"
-local CargBagsMixin = {}
+local KkthnxUIMixin = {}
 
-function CargBagsMixin:GetName()
+function KkthnxUIMixin:GetName()
     return addonName
 end
 
-function CargBagsMixin:Init()
-    return { "PLAYER_LOGIN" }
+function KkthnxUIMixin:Init()
+    return {"PLAYER_LOGIN"}
 end
 
-function CargBagsMixin:PLAYER_LOGIN()
-    local impl = cargBags:GetImplementation("KKUI_Backpack")
-    if impl then
-        hooksecurefunc(impl, "UpdateSlot", function(...) self:UpdateSlot(...) end)
+function KkthnxUIMixin:PLAYER_LOGIN()
+    local KkthnxUIBags = KkthnxUI[1].cargBags:GetImplementation("KKUI_Backpack")
+    if KkthnxUIBags then
+        hooksecurefunc(KkthnxUIBags, "UpdateSlot", function(...) self:UpdateSlot(...) end)
     end
 end
 
-function CargBagsMixin:SetTooltipItem(tooltip, item, locationInfo)
+function KkthnxUIMixin:SetTooltipItem(tooltip, item, locationInfo)
 	if locationInfo.bag == BANK_CONTAINER then
 		local hasItem, hasCooldown, repairCost, speciesID, level, breedQuality, maxHealth, power, speed, name = tooltip:SetInventoryItem("player", BankButtonIDToInvSlotID(locationInfo.slot))
 	else
 		local hasCooldown, repairCost, speciesID, level, breedQuality, maxHealth, power, speed, name = tooltip:SetBagItem(locationInfo.bag, locationInfo.slot)
 	end
 end
 
-function CargBagsMixin:Refresh()
-	-- TODO: This was missing
+function KkthnxUIMixin:Refresh()
+	KkthnxUI[1]:GetModule("Bags"):UpdateAllBags()
 end
 
-function CargBagsMixin:UpdateSlot(frame, bagID, slotID)
+function KkthnxUIMixin:UpdateSlot(frame, bagID, slotID)
+	local iconSize = KkthnxUI[2].Inventory.IconSize * 0.6
 	local button = frame:GetButton(bagID, slotID)
 	local options = {
-		showMogIcon=true, 
-		showBindStatus=true,
-		showSellables=true,
-		iconPosition="TOPRIGHT" 
+		showMogIcon = true,
+		showBindStatus = true,
+		showSellables = true,
+		iconPosition = "TOPRIGHT",
+		statusProminentSize = iconSize,
 	}
 
 	if button then
 		local item = CaerdonItem:CreateFromBagAndSlot(bagID, slotID)
-		CaerdonWardrobe:UpdateButton(button, item, self, { bag = bagID, slot = slotID }, options)
+		CaerdonWardrobe:UpdateButton(button, item, self, {bag = bagID, slot = slotID}, options)
 	end
 end
 
-local Version = nil
 if select(4, GetAddOnInfo(addonName)) then
-    if IsAddOnLoaded(addonName) then
-        local global = _G[addonName]
-		-- local global = GetAddOnMetadata(addonName, "X-cargBags")
-        if global then
-            -- cargBags = _G[global]
-            cargBags = global.cargBags
-		end
-
-		Version = GetAddOnMetadata(addonName, "Version")
-		if not Version then
-			Version = "Unknown"
-		end
-		if cargBags then
-			CaerdonWardrobe:RegisterFeature(CargBagsMixin)
+	if IsAddOnLoaded(addonName) then
+		if KkthnxUI[2].Inventory.Enable then
+			Version = GetAddOnMetadata(addonName, 'Version')
+			CaerdonWardrobe:RegisterFeature(KkthnxUIMixin)
```

Comment:
One possible issue here - if this can be toggled at game time, the functionality wouldn't get hooked until a reload.

Reply:
What would work better for you? Checking for carg being there instead like you have it previous? Maybe checking KkthnxUI then carg? Hmm.

Reply:
So typically I would register the feature because it's available but perhaps not enabled (but could be via config).  I think in most bag addon cases, I can just hook into them normally, but they don't fire their methods if they're not hooked up, so it doesn't result in anything out of the ordinary (so I can basically have all bag addons installed and they'll react appropriately based on which ones I have on / off or handling guild / bank / bags, etc.).  Was it causing some bad behavior if the bags were disabled?  If so, we'll probably just need to check if the bags are enabled before setting up the hook and then probably tying into some event to re-hook if they get enabled...

Reply:
Nothing was wrong. I just felt the need to contribute in some way. Though if you think this is unnecessary by all means you can deny it. Though we can implement so of the options like icon size and update all bags and such if you would like.

Reply:
Ok cool - I'd say we keep the icon size / update all bags improvements but drop the hook changes!

Reply:
I updated the PR. Thank you so much again for the help and support here. You are awesome.